internal/pkg/cli/env_delete.go

```diff
@@ -54,47 +53,72 @@ type DeleteEnvOpts struct {
 	deployClient environmentDeployer
 	prog         progress
 
-	// Cached objects to avoid multiple requests.
-	env *archer.Environment
-
 	*GlobalOpts
 }
 
-// Ask prompts for fields that are required but not passed in.
-func (opts *DeleteEnvOpts) Ask() error {
-	if opts.EnvName == "" {
-		envName, err := opts.prompt.Get(envDeleteNamePrompt, envDeleteNameHelpPrompt, validateEnvironmentName)
-		if err != nil {
-			return fmt.Errorf("prompt to get environment name: %w", err)
-		}
-		opts.EnvName = envName
-	}
-	if opts.EnvProfile == "" {
-		profile, err := opts.prompt.Get(
-			fmt.Sprintf(fmtEnvDeleteProfilePrompt, color.HighlightUserInput(opts.EnvName)),
-			envDeleteProfileHelpPrompt,
-			nil, // no validation needed
-			prompt.WithDefaultInput("default"))
-		if err != nil {
-			return fmt.Errorf("prompt to get the profile name: %w", err)
+// Validate returns an error if the individual user inputs are invalid.
+func (o *DeleteEnvOpts) Validate() error {
+	if o.EnvName != "" {
+		if err := o.validateEnvName(); err != nil {
+			return err
 		}
-		opts.EnvProfile = profile
 	}
 	return nil
 }
 
-// Validate returns an error if the environment name does not exist in the project, or if there are live applications under the environment.
-func (opts *DeleteEnvOpts) Validate() error {
-	// Validate that the environment is stored in SSM.
-	env, err := opts.storeClient.GetEnvironment(opts.ProjectName(), opts.EnvName)
+// Ask prompts for fields that are required but not passed in.
+func (o *DeleteEnvOpts) Ask() error {
+	if err := o.askEnvName(); err != nil {
+		return err
+	}
+	if err := o.askProfile(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Execute deletes the environment from the project by first deleting the stack and then removing the entry from the store.
+// If an operation fails, it moves on to the next one instead of halting the execution.
+// The environment is removed from the store only if other delete operations succeed.
+// Execute assumes that Validate is invoked first.
+func (o *DeleteEnvOpts) Execute() error {
+	if err := initDeleteEnvClientsFromProfile(o); err != nil {
+		return nil
+	}
+	if err := o.validateNoRunningApps(); err != nil {
+		return err
+	}
+
+	shouldDelete, err := o.shouldDelete(o.ProjectName(), o.EnvName)
 	if err != nil {
-		return fmt.Errorf("get environment %s metadata in project %s: %w", opts.EnvName, opts.ProjectName(), err)
+		return err
+	}
+	if !shouldDelete {
```

Comment:
nit: probably it would be better to print sth like `Deletion aborted by the user` before returning.